v2/tests/log_sink_etw_2_ut/log_sink_etw_2_ut.c

```diff
@@ -0,0 +1,802 @@
+
+    setup__tlgWriteTransfer_EventWriteTransfer(expected_event_metadata, 6, expected_event_data_descriptors, /* ignore file and func data */true);
+
+    // the actual event
+    setup__tlgCreate1Sz_char(); // message
+    setup__tlgCreate1Sz_char(); // file 
+    setup__tlgCreate1Sz_char(); // func
+    setup_EventDataDescCreate(); // line
+    uint8_t extra_metadata_bytes_2[256];
+    uint8_t expected_event_bytes_2[sizeof(SELF_DESCRIBED_EVENT) + sizeof(extra_metadata_bytes_2)];
+    SELF_DESCRIBED_EVENT* expected_event_metadata_2 = (SELF_DESCRIBED_EVENT*)&expected_event_bytes_2[0];
+    (void)memset(expected_event_metadata_2, 0, sizeof(SELF_DESCRIBED_EVENT));
+    expected_event_metadata_2->_tlgLevel = TRACE_LEVEL_CRITICAL;
+    expected_event_metadata_2->_tlgChannel = 11;
+    expected_event_metadata_2->_tlgOpcode = 0;
+    expected_event_metadata_2->_tlgKeyword = 0;
+    pos = (expected_event_bytes_2 + sizeof(SELF_DESCRIBED_EVENT));
+    // event name
+    (void)memcpy(pos, "LogCritical", strlen("LogCritical") + 1);
+    pos += strlen("LogCritical") + 1;
+    // content field
+    (void)memcpy(pos, "content", strlen("content") + 1);
+    pos += strlen("content") + 1;
+    *pos = TlgInANSISTRING;
+    pos++;
+    // content field
+    (void)memcpy(pos, "file", strlen("file") + 1);
+    pos += strlen("file") + 1;
+    *pos = TlgInANSISTRING;
+    pos++;
+    // content field
+    (void)memcpy(pos, "func", strlen("func") + 1);
+    pos += strlen("func") + 1;
+    *pos = TlgInANSISTRING;
+    pos++;
+    // content field
+    (void)memcpy(pos, "line", strlen("line") + 1);
+    pos += strlen("line") + 1;
+    *pos = TlgInINT32;
+    pos++;
+
+    expected_event_metadata_2->_tlgEvtMetaSize = (uint16_t)(pos - (expected_event_bytes_2 + sizeof(SELF_DESCRIBED_EVENT))) + 4;
+
+    captured_line = __LINE__;
+
+    // construct event data descriptor array
+    EVENT_DATA_DESCRIPTOR expected_event_data_descriptors_2[6] =
+    {
+        { 0 },
+        { 0 },
+        {.Size = (ULONG)strlen("test") + 1, .Ptr = (ULONGLONG)"test" },
+        {.Size = (ULONG)strlen(__FILE__) + 1, .Ptr = (ULONGLONG)__FILE__ },
+        {.Size = (ULONG)strlen(__FUNCTION__) + 1, .Ptr = (ULONGLONG)__FUNCTION__ },
+        {.Size = sizeof(int32_t), .Ptr = (ULONGLONG)&captured_line}
+    };
+
+    setup__tlgWriteTransfer_EventWriteTransfer(expected_event_metadata_2, 6, expected_event_data_descriptors_2, false);
+
+    expected_calls[3].u._get_pgmptr_call.override_result = true;
+    expected_calls[3].u._get_pgmptr_call.call_result = 1;
+
+    // act
+    log_sink_etw.log_sink_log(LOG_LEVEL_CRITICAL, NULL, __FILE__, __FUNCTION__, captured_line, "test");
+
+    // assert
+    POOR_MANS_ASSERT(expected_call_count == actual_call_count);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* very "poor man's" way of testing, as no test harness and mocking framework are available */
+int main(void)
+{
+    when__get_pgmptr_fails_log_sink_etw_log_prints_executable_as_UNKNOWN();
```

Comment:
>when__get_pgmptr_fails_log_sink_etw_log_prints_executable_as_UNKNOWN [](http://example.com/codeflow?start=4&length=68)

So this whole file is intended to test "what happens when we cannot get the executable name"?

... crazy!

Reply:
Yes :-).